examples/knowledge_tuning/knowledge_utils.py

```
@@ -97,10 +97,56 @@ def _conv_pretrain(rec):
+        entry = ds[i].copy()
+        if doc_count[doc] <= keep_no_qa_per_doc:
+            entry["unmask"] = True
+            unmask_entries.append(entry)
+        else:
+            entry["unmask"] = False
+            mask_entries.append(entry)
+            
+    ds_new = concatenate_datasets([Dataset.from_list(unmask_entries), Dataset.from_list(mask_entries)])
+    return ds_new
 
 def generate_knowledge_qa_dataset(
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)